pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -397,14 +400,30 @@ func (asw *actualStateOfWorld) GetVolumeMountState(volumeName v1.UniqueVolumeNam
 	return podObj.volumeMountStateForPod
 }
 
+// SetVolumeClaimSize sets actual pvc claim size in actualStateOfWorld for volume expansion.
+// Usually claimStatusSize is set when PVC enters actualStateOfWorld of the world but in case of kubelet
+// restart we might need to set claimStatusSize after the fact.
+func (asw *actualStateOfWorld) SetVolumeClaimSize(volumeName v1.UniqueVolumeName, claimSize resource.Quantity) {
+	asw.Lock()
+	defer asw.Unlock()
+
+	volumeObj, ok := asw.attachedVolumes[volumeName]
+	// only set volume claim size if claimStatusSize is zero
+	// this can happen when volume was rebuilt after kubelet startup
+	if ok && volumeObj.claimStatusSize.IsZero() {
```

Comment:
What happens if you expand a volume twice?

Reply:
if you expand a volume twice then only size recorded in DSOW changes until `NodeExpandVolume` is called and at which point size recorded in ASOW s updated as well.

Reply:
Not related to this change, but it may be worth thinking of a scenario where someone increases the size twice in quick succession. Would it be possible that we resize the fs for the first expansion, but when we update the pvc status, it updates the size to the 2nd value? (maybe an e2e test could be valuable too)

Reply:
yes If someone increases the size in quick succession and assuming control-plane is finished doing its thing, then size of PV will be second value. But in this case when `NodeExpandVolume` is invoked, it will get 2nd value recorded in PV as newSize (and not the intermediate one) and hence `pvc.Status.Capacity` will be updated with 2nd value.

Reply:
Do we need to think about a way we can address this? Such as recording the size before we call node expansion and using that value to update? This is kind of related to the challenge that the driver doesn't actually tell us the size.